code/Items/Data/Weapon/ItemWeapon.cs

```diff
@@ -0,0 +1,123 @@
+﻿using Sandbox;
+using ssl.Player;
+
+namespace ssl.Items.Data.Weapon
+{
+    public partial class ItemWeapon : Item
+    {
+        private const float PrimaryRate = 5.0F;
+
+        public ItemWeapon(string id, string name, string model) : base(id, name, model)
+        {
+        }
+
+        public override string ViewModelPath => "weapons/rust_pistol/v_rust_pistol.vmdl";
+
+        public override void OnInit(ItemStack itemStack)
+        {
+            base.OnInit(itemStack);
+            itemStack.Data = new WeaponData();
+        }
```

Comment:
don't let the item know the itemstack

Reply:
[https://github.com/Prakkkmak/Space-Station-Lambda/pull/219#discussion_r682682533](url)